examples/custom_methods.rs

```diff
@@ -0,0 +1,64 @@
+#![deny(warnings)]
+use std::net::SocketAddr;
+use std::str::FromStr;
+
+use warp::hyper::StatusCode;
+use warp::{hyper::Method, reject, Filter, Rejection, Reply};
+
+#[derive(Debug)]
+struct MethodError;
+impl reject::Reject for MethodError {}
+
+const FOO_METHOD: &'static str = "FOO";
+const BAR_METHOD: &'static str = "BAR";
+
+fn method(name: &'static str) -> impl Filter<Extract = (), Error = Rejection> + Clone {
+    warp::method()
+        .and_then(move |m: Method| {
+            async move {
```

Comment:
Looks like rustfmt is suggesting to move the `async move {` up to be on the same line as the start of the closure.

Reply:
Interesting, it seems like a missing `Cargo.toml` made my rustfmt fail in applying the correct formatting. Fixed by running it with the corresponding option manually.